vortex-array/src/compute/take.rs

```diff
@@ -79,19 +79,8 @@ impl ComputeFnVTable for Take {
 
         // We know that constant array don't need stats propagation, so we can avoid the overhead of
         // computing derived stats and merging them in.
-        if !taken_array.is::<ConstantVTable>() {
-            let derived_stats = derive_take_stats(array);
-
-            // TODO(robert): Ideally, we want to have a `combine_sets` method available on a
-            // `StatsSetRef` so we don't have to incur a clone here in `.to_owned()`.
-            let mut stats = taken_array.statistics().to_owned();
-            stats.combine_sets(&derived_stats, array.dtype())?;
-
-            for (stat, val) in stats {
-                // Alternatively, use a monoidal pattern here to set `stat = val`, or if it already
-                // exists, combine the two stats (similar to how `combine_sets` does it).
-                taken_array.statistics().set(stat, val)
-            }
+        if !taken_array.is_constant() {
```

Comment:
this shows up as a regression because we do extra work here now but I think this is the logic we want. I know I went back and forth on this